### PR TITLE
HAAR-1452 Call to get userId doesn't need to call nomis if already in auth database

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt [hunk 1]
```diff
@@ -151,6 +151,21 @@ class UserService(
       }
     }
 
+  @Transactional
+  fun getOrCreateUser(username: String, source: AuthSource): Optional<User> =
+    findUser(username, source).or {
+      getMasterUserPersonDetails(username, source).map {
+        val user = it.toUser()
+        if (AuthSource.fromNullableString(user.authSource) == nomis) {
+          getEmailAddressFromNomis(username).ifPresent { email ->
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt:160]: getEmailAddressFromNomis is common method  for other services and does additional call to "nomisUserService.getNomisUserByUsername(username)" can we move this nomis specific logic here 
 if (isHmpsGsiEmail(it)) Optional.empty() else Optional.of(it)

Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt:160]: I totally agree - but this is done in so many places.  Are you happy if I raise this as a separate ticket?
@ravmoj

Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt:160]: sure @karenmillermoj

**Review by ravmoj — APPROVED** [review]
lgtm

**Pull request merged**